codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -886,6 +895,16 @@ release name = execute_ (fromString $ "RELEASE " ++ name)
 rollbackTo name = execute_ (fromString $ "ROLLBACK TO " ++ name)
 rollbackRelease name = rollbackTo name *> release name
 
+withSavepoint :: (MonadUnliftIO m, DB m) => String -> (m () -> m r) -> m r
```

Comment:
We'll want to switch over to `unison-sqlite` of course, but that's a bigger job than would fit in this PR; we can't swap it in ad-hoc since the Connection object is a different type.